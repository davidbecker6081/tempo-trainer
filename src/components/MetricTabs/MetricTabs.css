.tab-container {
  display: flex;
  justify-content: space-between;
  margin-bottom: 1em;
  width: 60vw;
}

.tab {
  width: 16%;
  background-color: #FFFFFF;
  border-radius: 2px;
  box-shadow: 0px 1px 2px 2px rgba(82, 136, 155, 0.5);
  outline: none;
}

.tab:hover {
  cursor: pointer;
  border: 2px solid #2D2D2D;
}

.tab-active,
.tab-active:hover {
  background: #2D2D2D;
  border: 2px solid #2D2D2D;
  color: #FFFFFF;
  cursor: default;
}
