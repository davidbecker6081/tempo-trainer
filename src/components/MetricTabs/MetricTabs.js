import React from 'react';
import PropTypes from 'prop-types';
import './MetricTabs.css';

const MetricTabs = ({ handleChannelChange }) =>
  (
    <div>
      <button onClick={e => handleChannelChange('power')}>Power</button>
      <button onClick={e => handleChannelChange('heartRate')}>HeartRate</button>
      <button onClick={e => handleChannelChange('cadence')}>Cadence</button>
      <button onClick={e => handleChannelChange('temperature')}>Temperature</button>
      <button onClick={e => handleChannelChange('elevation')}>Elevation</button>
      <button onClick={e => handleChannelChange('speed')}>Speed</button>
    </div>
  );

export default MetricTabs;

MetricTabs.propTypes = {
  handleChannelChange: PropTypes.func.isRequired,
};
