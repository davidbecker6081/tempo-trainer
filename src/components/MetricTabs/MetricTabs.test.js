import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import MetricTabs from './MetricTabs';

describe('MetricTabs', () => {
  let wrapper;
  let mockProps;

  beforeEach(() => {
    mockProps = {
      handleChannelChange: () => {},
    };
  });

  it('should render correclty', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
