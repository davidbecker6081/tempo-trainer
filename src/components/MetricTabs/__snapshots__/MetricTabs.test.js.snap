// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MetricTabs should render correclty 1`] = `undefined`;
